pkg/attestation/crafter/runners/oidc/github.go

```diff
@@ -0,0 +1,271 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The code is a modified version of the code from the SLSA GitHub generator
+// https://github.com/slsa-framework/slsa-github-generator.
+
+package oidc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"slices"
+
+	"bytes"
+	"io"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+)
+
+var defaultActionsProviderURL = "https://token.actions.githubusercontent.com"
+var defaultAudience = []string{"nobody"}
+
+const (
+	requestTokenEnvKey = "ACTIONS_ID_TOKEN_REQUEST_TOKEN"
+	requestURLEnvKey   = "ACTIONS_ID_TOKEN_REQUEST_URL"
+)
+
+type GitHubOIDCClient struct {
+	requestURL   *url.URL
+	verifierFunc func(context.Context) (*oidc.IDTokenVerifier, error)
+	bearerToken  string
+	audience     []string
+	token        *GitHubToken
+}
+
+// Token represents the contents of a GitHub OIDC JWT token.
+type GitHubToken struct {
+	oidc.IDToken
+
+	// JobWorkflowRef is a reference to the current job workflow.
+	JobWorkflowRef string `json:"job_workflow_ref"`
+
+	// RunnerEnvironment is the environment the runner is running in.
+	RunnerEnvironment string `json:"runner_environment"`
+
+	// RawToken is the unparsed oidc token.
+	RawToken string
+}
+
+// NewOIDCGitHubClient returns new GitHub OIDC provider client.
+func NewOIDCGitHubClient() (Client, error) {
```

Comment:
this should return a concrete type not an interface https://duncanleung.com/go-idiom-accept-interfaces-return-types/